pkg/apis/kops/validation/legacy.go

```diff
@@ -400,6 +400,34 @@ func ValidateCluster(c *kops.Cluster, strict bool) field.ErrorList {
 		}
 	}
 
+	if c.Spec.IAMRolesForServiceAccounts != nil {
+		allErrs = append(allErrs, validateIRSA(c.Spec.IAMRolesForServiceAccounts, fieldSpec.Child("iamRolesForServiceAccounts"))...)
+	}
+	return allErrs
+}
+
+func validateIRSA(irsa *kops.IAMRolesForServiceAccountsConfig, spec *field.Path) (allErrs field.ErrorList) {
+	if !featureflag.UseServiceAccountIAM.Enabled() {
+		allErrs = append(allErrs, field.Forbidden(spec, "IAM roles for ServiceAccounts requires UseServiceAccountIAM feature flag to be enabled"))
+	}
+
+	sas := make(map[string]string)
+	for _, sa := range irsa.ServiceAccounts {
+		p := spec.Child("serviceAccounts")
+		key := fmt.Sprintf("%s/%s", sa.Namespace, sa.Name)
+		_, duplicate := sas[key]
+		if duplicate {
+			allErrs = append(allErrs, field.Duplicate(p, key))
+		}
+		sas[key] = ""
+
+		if sa.IAMPolicyARN == "" && sa.InlinePolicy == "" {
+			allErrs = append(allErrs, field.Required(p, "inlinePolicy or iamPolicyARN must be set"))
+		}
+		if sa.IAMPolicyARN != "" && sa.InlinePolicy != "" {
+			allErrs = append(allErrs, field.Forbidden(p, "cannot set both inlinePolicy and iamPolicyARN"))
+		}
```

Comment:
nit, might be worth validating the policy arn:

https://github.com/kubernetes/kops/blob/b03b50fdb14d5c44dec0144c17e2187425ffb1cd/pkg/apis/kops/validation/instancegroup.go#L267-L271